influxql/parser.go

```diff
@@ -771,38 +845,39 @@ func (p *Parser) parseCreateContinuousQueryStatement() (*CreateContinuousQuerySt
 	}
 
 	// Read the id of the query to create.
-	tok, pos, lit := p.scanIgnoreWhitespace()
-	if tok != IDENT && tok != STRING {
-		return nil, newParseError(tokstr(tok, lit), []string{"identifier", "string"}, pos)
+	ident, err := p.parseIdentifier()
+	if err != nil {
+		return nil, err
 	}
-	stmt.Name = lit
+	stmt.Name = ident
 
-	// Expect an "AS SELECT" keyword.
-	if tok, pos, lit := p.scanIgnoreWhitespace(); tok != AS {
-		return nil, newParseError(tokstr(tok, lit), []string{"AS"}, pos)
+	// Expect an "ON" keyword.
+	if tok, pos, lit := p.scanIgnoreWhitespace(); tok != ON {
+		return nil, newParseError(tokstr(tok, lit), []string{"ON"}, pos)
 	}
-	if tok, pos, lit := p.scanIgnoreWhitespace(); tok != SELECT {
-		return nil, newParseError(tokstr(tok, lit), []string{"SELECT"}, pos)
+
+	// Read the name of the database to create the query on.
+	if ident, err = p.parseIdentifier(); err != nil {
+		return nil, err
+	}
+	stmt.DB = ident
+
+	// Expect a "BEGIN SELECT" tokens.
+	if err := p.parseTokens([]Token{BEGIN, SELECT}); err != nil {
+		return nil, err
 	}
 
 	// Read the select statement to be used as the source.
-	source, err := p.parseSelectStatement()
+	source, err := p.parseSelectStatement(targetRequired)
 	if err != nil {
 		return nil, err
 	}
 	stmt.Source = source
 
-	// Expect an INTO keyword.
-	if tok, pos, lit := p.scanIgnoreWhitespace(); tok != INTO {
-		return nil, newParseError(tokstr(tok, lit), []string{"INTO"}, pos)
-	}
-
-	// Read the target of the query.
-	tok, pos, lit = p.scanIgnoreWhitespace()
-	if tok != IDENT && tok != STRING {
-		return nil, newParseError(tokstr(tok, lit), []string{"identifier", "string"}, pos)
+	// Expect a "END" keyword.
+	if tok, pos, lit := p.scanIgnoreWhitespace(); tok != END {
+		return nil, newParseError(tokstr(tok, lit), []string{"END"}, pos)
```

Comment:
It was mainly for tokens like `COMMA` or `WS` that have representations that are confusing in the final error message. I'd actually like to switch the error messages to have a more user friendly message. Parsing expressions is an example. It returns back a list of all the tokens (string, number, duration, etc) when the error message should just say it expected an "expression".